dashboard/src/components/ListingItem/ListingItem.tsx

```diff
@@ -90,20 +93,34 @@ const ListingItem = ({
   );
 
   const WrapperComponent = onClick ? 'button' : 'div';
+  const TooltipTriggerComponent = tooltip ? TooltipTrigger : Fragment;
 
   return (
-    <WrapperComponent
-      className={classNames('flex flex-row items-center gap-2 pb-1', hasBorder)}
-      onClick={handleOnClick}
-    >
-      {itemError}
-      {itemWarning}
-      {itemSuccess}
-      {itemUnknown}
-      {itemNeutral}
-      {leftIcon}
-      <span className="truncate text-sm text-black">{text}</span>
-    </WrapperComponent>
+    <Tooltip>
+      <div className="flex">
+        <TooltipTriggerComponent className="overflow-hidden">
+          <WrapperComponent
+            className={classNames(
+              'flex flex-row items-center gap-2 pb-1',
+              hasBorder,
+            )}
+            onClick={handleOnClick}
+          >
+            {itemError}
+            {itemWarning}
+            {itemSuccess}
+            {itemUnknown}
+            {itemNeutral}
+            {leftIcon}
+            <span className="truncate text-sm text-black">{text}</span>
+          </WrapperComponent>
+        </TooltipTriggerComponent>
+
+        <TooltipContent>
+          <span className="text-sm text-black">{text}</span>
+        </TooltipContent>
+      </div>
+    </Tooltip>
   );
```

Comment:
Nit: the less html/components we render the better, so i would only use the `Tooltip`, `TooltipTrigger`  and `TooltipContent` components when they are really needed.